### PR TITLE
fix: Make the entire advanced mode toggle container clickable

### LINKED ISSUE #7756: make the whole advanced mode container clickable
# Current behavior

At the bottom of the Settings sidebar, I have to click precisely on the toggle to activate Advanced mode.

# Desired Behavior

Make the click available on the whole Advanced Mode container. 

CleanShot 2024-10-16 at 16 55 12

### PR DESCRIPTION
**Pull request by Vardhaman619**
In this PR:

- Use a real `<input type="checkbox" />` element in the `<Toggle />` component
- Create an `accessibility` module in the `twenty-ui` package
- Export the `VISIBILITY_HIDDEN` CSS object to hide visually any element
- Export a `<VisibilityHidden />` component from the `twenty-ui` package to add visually hidden textual information easily
- Export a `<VisibilityHiddenInput />` component to create custom form control components easily
- Use a `<label>` element for the "Advanced:" text; it will naturally toggle the advanced settings

Fixes #7756

### CODE REVIEW

File: packages/twenty-front/src/modules/ui/navigation/link/components/AdvancedSettingsToggle.tsx [hunk 1]
```diff
@@ -46,13 +48,17 @@ export const AdvancedSettingsToggle = () => {
...
+  const toggleAdvancedMode = () => {
+    setIsAdvancedModeEnabled((current) => !current);
+  };
+
   return (
     <StyledContainer>
       <StyledIconContainer>
         <StyledIconTool size={12} color={MAIN_COLORS.yellow} />
       </StyledIconContainer>
       <StyledToggleContainer>
```
Comment (charlesBochet) [inline packages/twenty-front/src/modules/ui/navigation/link/components/AdvancedSettingsToggle.tsx:60]: I think we should make the Container clickable
@Devessier what's your recommendation here ? use a button?

I feel the "Advanced" is actually a label for the toggle

Comment (Devessier) [reply packages/twenty-front/src/modules/ui/navigation/link/components/AdvancedSettingsToggle.tsx:60]: Thanks for mentioning me, @charlesBochet.

Indeed, I think we should make the text "Advanced:" a true HTML `<label>` element. I don't believe the whole container should be clickable, as clicking on a blank space would update the toggle, which might seem strange.
The issue is that the `<Toggle />` component is only made of `<div>` and not a proper `<input>` element, so using a `<label>` would do nothing. I had this component on my list of accessibility things to discuss.

See https://github.com/twentyhq/twenty/blob/05e8f8a0b1fd4c0b1e53c289baba3d073565b61a/packages/twenty-front/src/modules/ui/input/components/Toggle.tsx

Comment (charlesBochet) [reply packages/twenty-front/src/modules/ui/navigation/link/components/AdvancedSettingsToggle.tsx:60]: @Vardhaman619 could you update our Toggle component to include a <input>
Take inspiration from chakra ui regarding the html structure
image

It looks to me that they have an invisible input

Comment (Devessier) [reply packages/twenty-front/src/modules/ui/navigation/link/components/AdvancedSettingsToggle.tsx:60]: Please note that the styles applied to hide the `input` are to make it visible only for screen readers (the tool used by people with disabilities).

These styles are now normalized and everybody uses the same to hide an element for people not using a screen reader:

```css
// In Tailwind:

.sr-only {
  position: absolute;
  width: 1px;
  height: 1px;
  padding: 0;
  margin: -1px;
  overflow: hidden;
  clip: rect(0, 0, 0, 0);
  white-space: nowrap;
  border-width: 0;
}
```

Some resources : 

- https://webaim.org/techniques/css/invisiblecontent/
- https://tailwindcss.com/docs/screen-readers

Comment (charlesBochet) [reply packages/twenty-front/src/modules/ui/navigation/link/components/AdvancedSettingsToggle.tsx:60]: We can introduce a <InvisibleInput /> component in this case, let's put it in an accessibility folder in twenty-ui

Comment (Devessier) [reply packages/twenty-front/src/modules/ui/navigation/link/components/AdvancedSettingsToggle.tsx:60]: Chakra UI has two components : VisibilityHidden and VisibilityHiddenInput. (https://v2.chakra-ui.com/docs/components/visually-hidden/usage#visually-hidden-input)

Their VisibilityHidden component renders a `<span>` as it's usually used to provide little indications to the users, like a hidden label explaining what shows an icon.

**Comment by Devessier** [comment]
I'm working on it.

**Comment by Devessier** [comment]
Thank you a lot for your contribution, @Vardhaman619. It was a critical bug fix, and I handled the update.

**Review by charlesBochet — APPROVED** [review]

**Pull request merged**